### PR TITLE
babashka: 0.0.71 -> 0.0.78

### PR DESCRIPTION
**Pull request by bhougland18**
###### Motivation for this change
Add new package to repository.  This is a small command line utility that allows one to use a subset of the clojure programming language in their shell commands.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/babashka/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{ stdenv, lib, graalvm8, fetchurl }:
```
Comment (AndersonTorres) [inline pkgs/tools/misc/babashka/default.nix]: Normally we put fetchurl right after stdenv:

`{ stdenv, fetchurl, lib, graalvm8 }:`

File: pkgs/tools/misc/babashka/default.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      "-J-Xmx3g"
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp bb $out/bin/bb
+  '';
+
+  meta = with lib; {
+    description = "A Clojure babushka for the grey areas of Bash.";
```
Comment (AndersonTorres) [inline pkgs/tools/misc/babashka/default.nix]: No end point here.

`description = "A Clojure babushka for the grey areas of Bash";`

File: pkgs/tools/misc/babashka/default.nix [hunk 3]
```diff
@@ -0,0 +1,57 @@
+{ stdenv, fetchcurl, lib, graalvm8 }:
```
Comment (mmilata) [inline pkgs/tools/misc/babashka/default.nix]:
```suggestion
{ stdenv, fetchurl, lib, graalvm8 }:
```

**Comment by AndersonTorres** [comment]
Why is it failing the Borg?

File: pkgs/tools/misc/babashka/default.nix (hunk 3 above)
Comment (AndersonTorres) [inline pkgs/tools/misc/babashka/default.nix]:
```suggestion
{ stdenv, fetchurl, graalvm8 }:
```

File: pkgs/tools/misc/babashka/default.nix [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      --report-unsupported-elements-at-runtime \
+      "-J-Xmx3g"
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp bb $out/bin/bb
+  '';
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/tools/misc/babashka/default.nix]:
```suggestion
   meta = {
```

File: pkgs/tools/misc/babashka/default.nix [hunk 5]
```diff
@@ -0,0 +1,57 @@
+{ stdenv, fetchcurl, lib, graalvm8 }:
+
```
Comment (AndersonTorres) [inline pkgs/tools/misc/babashka/default.nix]:
```suggestion
with stdenv.lib;
```

**Comment by bandresen** [comment]
Maybe the location should be `pkgs/development/interpreters/clojure/babashka.nix` instead of `pkgs/tools/misc/babashka/default.nix`?

Anyway: looking forward to this package!

File: pkgs/tools/misc/babashka/default.nix [hunk 6]
```diff
@@ -37,19 +39,44 @@ stdenv.mkDerivation rec {
...
+    - Allow interop with commonly used classes like java.io.File and System
+    - Multi-threading support (pmap, future, core.async)
+    - Batteries included (tools.cli, cheshire, ...)
+    - Library support via popular tools like the clojure CLI
+    '';
+    homepage = "https://github.com/borkdude/babashka";
     license = licenses.epl10;
     platforms = graalvm8.meta.platforms;
-    maintainers = with maintainers; [ DerGuteMoritz ];
+    maintainers = with maintainers; [ bhougland jlesquembre DerGruteMoritz ];
```
Comment (bandresen) [inline pkgs/tools/misc/babashka/default.nix:80]: the spelling of this maintainer is DerGuteMoritz

**Comment by bandresen** [comment]
Not sure why eval fails by borg, but for me this PR builds fine and works as expected.

@mmilata @AndersonTorres

**Comment by bhougland18** [comment]
> Not sure why eval fails by borg, but for me this PR builds fine and works as expected.
> 
> @mmilata @AndersonTorres

I am currently using the software as well, works fine.

**Comment by bandresen** [comment]
I would really like to see the latest babashka in NixOS, I would like to propose the following:

I resubmit a new PR, basically re-doing #83886 with the, I think, proper location of `pkgs/development/interpreters/clojure/babashka.nix`, based on master. I would use this PR's meta.description and meta.longDescription.

This PR has all its commits removed sans adding @bhougland18 to the maintainer list which can get merged then.
Then I will add @DerGuteMoritz, @bhougland18 & myself to the maintainers of this package in my new PR, so we can all keep an eye on new releases, graalvm updates affecting the package, etc.

Can we agree on this course of action @AndersonTorres and @bhougland18?

**Comment by bhougland18** [comment]
I agree with this course of action.  This is my first commit and I really
don't know where to go from here as I don't understand why it is failing.

**Comment by AndersonTorres** [comment]
OK for me.

In fact I was uploading a new diff for it now.

P.S.: I am a bit puristic, and I would like to ast two things:

- Is it hard to compile directly from sources, instead of using a JAR file?
- Can it be renamed as "babashka-jar"?

**Comment by AndersonTorres** [comment]
"Superceded" by https://github.com/NixOS/nixpkgs/pull/84284

**Pull request closed without merging**